### ISSUE TITLE
incompatible types: 'Movie' cannot be explicitly typed as 'json' in example

### ISSUE THREAD

**Issue by mhmxs** [ballerina-platform/ballerina-lang#12922]
**Description:**
I tried to convert a record to a JSON by following the example [here](https://github.com/ballerina-platform/ballerina-lang/blob/90e7560b42ad249c87c95f5b2f1d1ff88d845455/examples/json-record-map-conversion/json_record_map_conversion.bal#L30)

**Steps to reproduce:**
Create a local file by downloading the [example](https://raw.githubusercontent.com/ballerina-platform/ballerina-lang/90e7560b42ad249c87c95f5b2f1d1ff88d845455/examples/json-record-map-conversion/json_record_map_conversion.bal).
Delete lines after line 32 (except the last line)
and than
```
ballerina build
Compiling source
    x.bal
error: .::x.bal:30:20: incompatible types: 'Movie' cannot be explicitly typed as 'json'
error: .::x.bal:30:20: invalid usage of the checked expression operator: no error type return in enclosing invokable
```

**Affected Versions:**
Ballerina 0.990.0

**OS, DB, other environment details and versions:**
macOS Mojave 10.14.1

**Comment by MaryamZi** [comment]
@mhmxs similar to #12921 this too is due to issues with the BBE, which we've fixed for the upcoming 0.990.1 release.

The updated example is available at  https://github.com/ballerina-platform/ballerina-lang/blob/v0.990.1/examples/json-record-map-conversion/json_record_map_conversion.bal.

Ballerina 0.990.0 introduced several changes, which result in these errors once the lines after L32 are removed:
- the diamond operator results in a type assertion (the diamond operator resulted in a conversion pre-0.990.0) 
- using the `check` operator returns the error if the type of the expression is error, and thus mandates the enclosing function to optionally return error (the `check` operator used to throw the error if the function did not say it could return an error pre-0.990.0)

The previous set of errors (with the code after L32) were due to the maps being unconstrained. From Ballerina 0.990.0 onward, the constraint is mandatory for maps.

Please let us know if you face further issues.

**Comment by mhmxs** [comment]
Thanks, for me it was two separated issue, sorry for duplicating them.
